### PR TITLE
Review Cloud Balancing Quickstart Chapter

### PR DESCRIPTION
**Pull request by gsheldon**
Completed review and suggested modularization structure of content for this chapter.

### CODE REVIEW

File: optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc [hunk 1]
```diff
@@ -1,5 +1,7 @@
+
 [[quickStart]]
-= Quick Start
+= Getting Started: A Cloud Balancing Demonstration
+//Christy suggested that this isn't really a "quickstart". I don't know how you feel about renaming a chapter?
```
Comment (ge0ffrey) [inline optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc]: Hey Gemma, can we put these sort of comments in the PR, but not in the actual source (so it doesn't get merged by accident)?

Can you explain why this isn't really a quickstart? We can rename it or restructure it if it's not correct, of course. I want to understand why/how. The goal of this chapter is for users to get started quickly by learning the minimal basics to implement an optaplanner case, without reading 350 pages of documentation (which they don't do).

File: optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc [hunk 2]
```diff
@@ -445,14 +467,18 @@ public class CloudBalancingEasyScoreCalculator implements EasyScoreCalculator<Cl
 ----
 ====
 
-Even if we optimize the code above to use ``Map``s to iterate through the `processList` only once, *it is still slow* because it does not do <<incrementalScoreCalculation,incremental score calculation>>.
+Even if we optimize the code above to use ``Map``s to iterate through the `processList` only once, it is still slow because it does not do <<incrementalScoreCalculation,incremental score calculation>>.
```
Comment (ge0ffrey) [inline optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc]: Please keep the boldness of "it is still slow". I've seen users claim optaplanner doesn't scale while using an easy score calculator because they don't realize that that is the problem. It needs to stand out.

File: optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc [hunk 3]
```diff
@@ -445,14 +467,18 @@ public class CloudBalancingEasyScoreCalculator implements EasyScoreCalculator<Cl
...
 To fix that, either use incremental Java score calculation or Drools score calculation.
-Let's take a look at the latter.
 
 
 [[cloudBalancingDroolsScoreConfiguration]]
 ==== Drools Score Configuration
 
+Drools score calculation uses incremental calculation, where every score constraint is written as one or more score rules. 
+
+If you use the Drools rule engine for score calculation, you can integrate with other Drools technologies, such as decision tables (XLS or web based), the KIE Workbench, ...
```
Comment (ge0ffrey) [inline optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc]: Shall we rephrase this to get rid of the "you use" and "you can"?

File: optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc [hunk 4]
```diff
@@ -355,7 +225,161 @@ public class CloudBalance ... {
...
+  </termination>
+</solver>
+----
+====
+
+This solver configuration consists of three parts:
+
+. **Domain model configuration**
+
+What can Planner change?::
```
Comment (ge0ffrey) [inline optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc]: There are changes in this big green blob, such as these ::, but I can't see them on github review... How shall we deal with this?

File: optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc [hunk 5]
```diff
@@ -41,10 +43,6 @@ As you can see, it is not optimal, as it does not leave enough room to assign th
 Planner does find the more optimal solution by using additional, smarter algorithms.
 It also scales: both in data (more processes, more computers) and constraints (more hardware requirements, other constraints). So let's see how Planner can be used in this scenario.
 
-
-[[cloudBalancingValueProposition]]
-=== Value Proposition
```
Comment (ge0ffrey) [inline optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc]: The examples in chapter 3 each have this value proposition section, so it shows up in the TOC. If it's removed here, it needs to be removed there too.

File: optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc [hunk 6]
```diff
@@ -71,160 +69,27 @@ image::Chapter-Quick_start/cloudOptimizationValueProposition.png[align="center"]
 [[cloudBalancingDomainModelDesign]]
 === Domain Model Design
 
-Beginning with the domain model:
 
+Using a <<domainModelingGuide,domain model>> helps determine which classes are planning entities and which of their properties are planning variables. Using a domain model helps to simplify contraints, improve performance, and increase flexibility for future needs.
```
Comment (ge0ffrey) [inline optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc]: "Using a domain model helps ..." => There is always a domain model, even if it's implicitly, it's not optional. 
change it to "Using a good domain model helps ..."

"Using a domain model helps ..." is stated twice.

File: optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc [hunk 7]
```diff
@@ -71,160 +69,27 @@ image::Chapter-Quick_start/cloudOptimizationValueProposition.png[align="center"]
...
 
-Beginning with the domain model:
 
+Using a <<domainModelingGuide,domain model>> helps determine which classes are planning entities and which of their properties are planning variables. Using a domain model helps to simplify contraints, improve performance, and increase flexibility for future needs. 
+
+//procedure from chapter 17 - Design Patterns
+
+. Draw a class diagram of your domain model. 
+. Normalize it to remove duplicate data.
+. Write down some sample instances for each class.
```
Comment (ge0ffrey) [inline optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc]: Then the next 3 lines should have some sample instances written down too :)

File: optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc [hunk 8]
```diff
@@ -71,160 +69,27 @@ image::Chapter-Quick_start/cloudOptimizationValueProposition.png[align="center"]
 [[cloudBalancingDomainModelDesign]]
 === Domain Model Design
 
-Beginning with the domain model:
 
+Using a <<domainModelingGuide,domain model>> helps determine which classes are planning entities and which of their properties are planning variables. Using a domain model helps to simplify contraints, improve performance, and increase flexibility for future needs. 
+
+//procedure from chapter 17 - Design Patterns
```
Comment (ge0ffrey) [inline optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc]: that's a subset of the procedure. No need to have the comment if the section above already links it.

This chapter definitely doesn't need the entire procedure - as this is a getting started quickly.
I was first not sure if it needs any of the procedure - we can maybe just present the solution and link to the procedure in chapter 17? OTH, I also see the value of keeping a subset procedure here - and especially asking the key question "what changes during planning"? So I agree that having a short subset procedure here has value.

File: optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc (hunk 1 above)
Comment (gsheldon) [inline optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc]: Hey Geoffrey, I asked Christy to clarify this, and the basic idea is that we (as in, CCS as a whole), are working towards more clear and consistent terminology and a standardization of how we use certain terms. "Quickstart", in our terminology, would refer to something that includes installation information to actually get a product or program up and running, whereas "Getting Started" focuses more on tutorials and how to actually use the product once you have already installed it and have it up and running.

File: optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc (hunk 5 above)
Comment (gsheldon) [inline optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc]: I have modified Chapter 3 to include the image itself as a temporary fix. I still need to include a paragraph to explain this slide, however I haven't really worked out how to do that yet.

File: optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc (hunk 6 above)
Comment (gsheldon) [inline optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc]: +1 Fixed.

File: optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc (hunk 8 above)
Comment (gsheldon) [inline optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc]: Part of the modularization strategy is to have everything the user needs to know to complete a task in one place. If we send them off to Chapter 17 to learn all about Design Patterns, we might not get them back. It's only a short part of the procedure that I've borrowed from there, so I'm hoping we can leave it in.

File: optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc (hunk 2 above)
Comment (gsheldon) [inline optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc]: Normally we wouldn't emphasise something in the middle of a sentence like this. Just from a stylistic point of view, it seems a little casual for a technical document. However, if you really want to emphasise it, the IBM Style Guide recommends italics.

File: optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc (hunk 4 above)
Comment (gsheldon) [inline optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc]: I've changed these to bold.

File: optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc (hunk 7 above)
Comment (gsheldon) [inline optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc]: I hope the change I have made here is right and makes sense. I just went from the image we are referring to for this section.

File: optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc (hunk 3 above)
Comment (gsheldon) [inline optaplanner-docs/src/main/asciidoc/Chapter-Quick_start/Chapter-Quick_start.adoc]: +1 made a change.

**Comment by ge0ffrey** [comment]
jenkins test this again

**Comment by ge0ffrey** [comment]
Merged. Sorry the delay in merging this - I hadn't seen that the feedback was fixed 6 days ago already.
